server/gc/src/main/java/org/apache/accumulo/gc/SimpleGarbageCollector.java

```diff
@@ -564,6 +566,19 @@ private void run() {
     ProbabilitySampler sampler =
         new ProbabilitySampler(getConfiguration().getFraction(Property.GC_TRACE_PERCENT));
 
+    // This is created outside of the run loop and passed to the walogCollector so that
+    // only a single timed task is created (internal to LiveTServerSet using SimpleTimer.
+    final LiveTServerSet liveTServerSet = new LiveTServerSet(this, new LiveTServerSet.Listener() {
+      @Override
+      public void update(LiveTServerSet current, Set<TServerInstance> deleted,
+          Set<TServerInstance> added) {
+        if (!deleted.isEmpty() || !added.isEmpty()) {
+          log.debug("Number of current servers {}, tservers added {}, removed {}",
+              current == null ? -1 : current.size(), added, deleted);
+        }
+      }
+    });
+
```

Comment:
Why not call here `liveServers.startListeningForTabletServerChanges()` here?  I don't think its needed if my other suggestion to call `scanServers` is made.  However I am curious if there was  reason for not calling it.

Reply:
I discussed this solution with @EdColeman yesterday. I think the intent was to call `liveServers.startListeningForTabletServerChanges()` here. I had thought it was present when I approved the PR, but now I can't find it in the commits.

Reply:
ok, I think I like calling `scanServers()` explicitly anyway because it seems easier to reason about when things happen during the GC cycle.  Otherwise there is some timer that runs concurrently at random points during the cycle.

Reply:
I agree.

Reply:
I'll put the call to scanServers - that's seems better than a random timer task update.  I'll keep the external tserver list that get passed to the wal collector so that it does update between gc runs as desired.

I'm also seeing what I can do about adding or modify the tests - the current tests pass in a mock instance and it they don't seem to exercise a path when the set changes.